build.sbt

```diff
@@ -530,6 +532,12 @@ lazy val buildInfoSettings = Def.settings(
   buildInfoObject := "BuildInfo"
 )
 
+lazy val noPublish = Seq(
+  publishArtifact := false,
+  publish := {},
+  publishLocal := {}
+)
```

Comment:
I suggest we deduplicate this with https://github.com/scalameta/scalameta/blob/d38a0f9cd5e58373967d31e2a9ebe1ee67662690/build.sbt#L50-L51.